### DISCUSSION TITLE
How should typing_extensions TypeVar default be used with a Generic?

### DISCUSSION THREAD

**Discussion by gandhis1** [python/typing#1518] [Q&A] [answered]
Example ([link to Mypy playground](https://mypy-play.net/?mypy=latest&python=3.11&gist=2313c1dbeb8c31f040f4abc203b4e97b&flags=strict)):

```python
from typing_extensions import Generic, TypeVar

T = TypeVar("T", default=int)

class ListWrapper(Generic[T]):
    def __init__(self, input_list: list[T]) -> None:
        self.input_list = input_list


def print_wrapper(wrapper_instance: ListWrapper) -> None:  # main.py:10: error: Missing type parameters for generic type "ListWrapper"  [type-arg]
    print(wrapper_instance.input_list)

x = ListWrapper([1, 2, 3])
reveal_type(x)  # main.py:14: note: Revealed type is "__main__.ListWrapper[builtins.int]"
print_wrapper(x)
```

I suspect I am just misunderstanding the way this works. I had the expectation that I would no longer need to specialize `ListWrapper`, and the omission of the `TypeVar` would be assumed to mean `ListWrapper[int]`. Could someone please clarify?

**Comment by Gobot1234** [comment, marked as answer]
This is a bug in mypy's implementation (pyright's works fine here https://pyright-play.net/?code=GYJw9gtgBALgngBwJYDsDmB9ApgDxllAZyTCKiQgTBBigHECsQkBjAGigBVEsA1AQxABYAFCjOUALxceAkAAoARJ0UcAJlmD8ArgBsYk1DACUo0S139ChKABkkhGAHUQ-BAibyGKJqwDanAC6xgBcolARUBrAUBgYqEgwcfKEWLrAHKgI2km6DjAhUHmOAcFQALQAfFAAcqRYYSKRzVCp6QB0WTkYxbTSXbn5ZmIi0VAIzChJAO6u7p6zbh4g8UQw-CgsDXb5LktMxhXVdT6FUADEUBD8qO0IcCEAjAAMhUzgIIUAsg7E6LA8caCfgQLD4EA2YDUKBoRjMFgAjxQRT2Rx7eYgRQRPzwDzlQRoQLhSITIzyRYY1aODZbToobKDRymEY4KQ7NFzZbyPyPDgAJg4AGZgqIQFgAG5Yfi6DC4rDyHCHC5XG4oO4PR4AFkKKDA+EKACUJVLdFg1IisOQbIo4tdUHF2qjnJymH4AEbaJD6VCEOkwQKKUSkqYYClcxVAA). It should work according to the PEP.

**Reply by erictraut** [reply]
I don't think it's fair to call this a "bug" in mypy. Mypy simply hasn't implemented support for PEP 696. This is understandable given that PEP 696 is still in draft form and hasn't been accepted by the steering council.

Pyright does have (provisional) support for the latest draft of PEP 696 if you want to play with it.

**Reply by gandhis1** [reply]
Thank you. Linking to the corresponding issue on Mypy's issue tracker for others' reference:

https://github.com/python/mypy/issues/14851
